examples/step-85/step-85.cc

```diff
@@ -0,0 +1,565 @@
+                                           invalid_subface);
+
+                const unsigned int n_interface_dofs =
+                  fe_interface_values.n_current_interface_dofs();
+                FullMatrix<double> local_stabilization(n_interface_dofs,
+                                                       n_interface_dofs);
+                for (unsigned int q = 0;
+                     q < fe_interface_values.n_quadrature_points;
+                     ++q)
+                  {
+                    const Tensor<1, dim> normal = fe_interface_values.normal(q);
+                    for (unsigned int i = 0; i < n_interface_dofs; ++i)
+                      for (unsigned int j = 0; j < n_interface_dofs; ++j)
+                        {
+                          local_stabilization(i, j) +=
+                            .5 * ghost_parameter * cell_side_length * normal *
+                            fe_interface_values.jump_in_shape_gradients(i, q) *
+                            normal *
+                            fe_interface_values.jump_in_shape_gradients(j, q) *
+                            fe_interface_values.JxW(q);
+                        }
+                  }
+
+                const std::vector<types::global_dof_index>
+                  local_interface_dof_indices =
+                    fe_interface_values.get_interface_dof_indices();
+
+                stiffness_matrix.add(local_interface_dof_indices,
+                                     local_stabilization);
+              }
+        }
+    }
+}
+
+
+// @sect3{Solving the System}
+template <int dim>
+void Step85<dim>::solve()
+{
+  std::cout << "Solving system" << std::endl;
+
+  const unsigned int max_iterations = solution.size();
+  SolverControl      solver_control(max_iterations);
+  SolverCG<>         solver(solver_control);
+  solver.solve(stiffness_matrix, solution, rhs, PreconditionIdentity());
+}
+
+
+
+// @sect3{Data Output}
+// Since we use separate DoFHandlers for the solution and the level set
+// function, it is easiest to write them to separate files: "solution.vtu" and
+// "level_set.vtu". Since both use the same triangulation, it is possible to
+// read both files in Visit/Paraview and combine the two files inside the
+// Visit/Paraview. This can, for example, be done using the "Append Attributes"
+// filter in Paraview.
+template <int dim>
+void Step85<dim>::output_results() const
+{
+  DataOut<dim> data_out;
+  data_out.attach_dof_handler(dof_handler);
+  data_out.add_data_vector(solution, "solution");
+  data_out.build_patches();
+  std::ofstream output("solution.vtu");
+  data_out.write_vtu(output);
+
+  DataOut<dim> data_out_level_set;
+  data_out_level_set.attach_dof_handler(level_set_dof_handler);
+  data_out_level_set.add_data_vector(level_set, "level_set");
+  data_out_level_set.build_patches();
+  std::ofstream output_level_set("level_set.vtu");
+  data_out_level_set.write_vtu(output_level_set);
+}
```

Comment:
You should be able to add different `DoFHandler`s to the same `DataOut` instance; you just need to call the `add_data_vector` versions that take `DoFHandler` as parameter. If it not working there is probably a bug.